restapi/src/main/java/io/stargate/web/docsapi/service/query/condition/ConditionProviderService.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.web.docsapi.service.query.condition;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.stargate.web.docsapi.exception.DocumentAPIRequestException;
+import io.stargate.web.docsapi.service.query.condition.provider.ConditionProvider;
+import io.stargate.web.docsapi.service.query.condition.provider.impl.BasicConditionProvider;
+import io.stargate.web.docsapi.service.query.condition.provider.impl.ExistsConditionProvider;
+import io.stargate.web.docsapi.service.query.condition.provider.impl.ListConditionProvider;
+import io.stargate.web.docsapi.service.query.filter.operation.impl.*;
+import java.util.*;
+
+/**
+ * Simple service that wraps all available raw filter values and connects them to a {@link
+ * ConditionProvider}.
+ *
+ * <p>Currently uses a unmodifiable map, thus it's thread safe.
+ */
+public class ConditionProviderService {
+
+  /** Map of raw filters to the Condition providers. */
+  private final Map<String, ConditionProvider> providerMap;
+
+  public ConditionProviderService() {
+    this.providerMap = initProviderMap();
+  }
+
+  /**
+   * Creates the conditions for the node containing the raw filter ops as the keys. For example:
+   * <code>{ "$gt: { 5 }, "$lt": { 10 }}</code>.
+   *
+   * @param conditionsNode Node containing the filter ops as keys
+   * @param numericBooleans If numeric boolean should be applied to the created conditions
+   * @return Collection of created conditions.
+   * @throws io.stargate.web.docsapi.exception.DocumentAPIRequestException If filter op is not
+   *     found, condition constructions fails or filter value is not supported by the filter op.
+   */
+  public Collection<BaseCondition> getConditions(JsonNode conditionsNode, boolean numericBooleans) {
```

Comment:
Could `numericBooleans` be an instance-level field in this class?

Reply:
ok, even better, removed from the expression service then..